src/Spryker/Client/RabbitMq/Model/Connection/ConnectionConfigMapper/ConnectionConfigMapper.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace Spryker\Client\RabbitMq\Model\Connection\ConnectionConfigMapper;
+
+use Spryker\Client\RabbitMq\Model\Exception\ConnectionConfigIsNotDefinedException;
+use Spryker\Client\RabbitMq\RabbitMqConfig;
+
+class ConnectionConfigMapper implements ConnectionConfigMapperInterface
+{
+    /**
+     * @var \Spryker\Client\RabbitMq\RabbitMqConfig
+     */
+    private $config;
+
+    /**
+     * @var \Generated\Shared\Transfer\QueueConnectionTransfer[]|null Keys are connection names.
+     */
+    protected $connectionsConfigMap;
+
+    /**
+     * @var \Generated\Shared\Transfer\QueueConnectionTransfer[]|null Keys are store names, values are lists of connections.
+     */
+    protected $connectionsConfigMapByStoreName;
+
+    /**
+     * @var \Generated\Shared\Transfer\QueueConnectionTransfer[]|null Keys are pool names, values are lists of connections.
+     */
+    protected $connectionsConfigMapByPoolName;
+
+    /**
+     * @param \Spryker\Client\RabbitMq\RabbitMqConfig $config
+     */
+    public function __construct(RabbitMqConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\QueueConnectionTransfer[][]
+     */
+    public function mapConnectionsConfigByStoreName(): array
+    {
+        if ($this->connectionsConfigMapByStoreName) {
+            return $this->connectionsConfigMapByStoreName;
+        }
+
+        $connectionConfigMap = [];
+        foreach ($this->getConnectionsConfigMap() as $connectionConfig) {
+            foreach ($connectionConfig->getStoreNames() as $storeName) {
+                $connectionConfigMap[$storeName][] = $connectionConfig;
+            }
+        }
+        $this->connectionsConfigMapByStoreName = $connectionConfigMap;
+
+        return $this->connectionsConfigMapByStoreName;
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\QueueConnectionTransfer[]
+     */
+    protected function getConnectionsConfigMap(): array
+    {
+        if ($this->connectionsConfigMap === null) {
+            $this->addConnectionsConfig();
+        }
+
+        return $this->connectionsConfigMap;
+    }
+
+    /**
+     * @return void
+     */
+    protected function addConnectionsConfig(): void
+    {
+        foreach ($this->config->getQueueConnections() as $queueConnectionConfig) {
+            $this->connectionsConfigMap[$queueConnectionConfig->getName()] = $queueConnectionConfig;
```

Comment:
Do we need this caching? We should avoid having states for models if there is no performance problem.

Reply:
Should models also be stateless? Didn't hear about it.

I kept this approach from the original `ConnectionManager`, I guess it was done for performance reasons, just to avoid recalculation. However, don't think it makes a big impact on performance.